specification/dartLangSpec.tex

```diff
@@ -1514,65 +1692,210 @@ \section{Variables}
 }
 
 \LMHash{}%
-% This error can occur due to implicit casts, and
-% for instance variables also when a setter is called dynamically.
+% This error can occur due to an implicit downcast from \DYNAMIC.
 It is a dynamic type error if the dynamic type of $o$ is not
-a subtype of the actual type of the variable $v$
+a subtype of the actual type of the variable \id{}
 (\ref{actualTypes}).
 
 
 \subsection{Evaluation of Implicit Variable Getters}
 \LMLabel{evaluationOfImplicitVariableGetters}
 
 \LMHash{}%
-\BlindDefineSymbol{d, v}%
-Let $d$ be the declaration of a static or instance variable $v$.
-If $d$ is an instance variable,
-then the invocation of the implicit getter of $v$ evaluates to
-the object stored in $v$.
-If $d$ is a static variable
-(\commentary{which can be a library variable})
-then the implicit getter method of $v$ executes as follows:
+We introduce two specific kinds of getters in order to specify
+the behavior of various kinds of variables
+without duplicating the specification of their common behaviors.
+
+\LMHash{}%
+A
+\IndexCustom{late-initialized getter}{getter!late-initialized}
+is a getter $g$ which is implicitly induced by a non-local variable $v$
+that has an initializing expression $e$.
+\commentary{%
+It is described below which declarations induce a late-initialized getter.%
+}
+An invocation of $g$ proceeds as follows:
+
+\LMHash{}%
+If the variable $v$ has not been bound to an object then
+$e$ is evaluated to an object $o$.
+If $v$ has now been bound to an object, and $v$ is immutable,
+a dynamic error occurs. % Implementations throw a 'LateInitializationError'.
+Otherwise, $v$ is bound to $o$,
+and the evaluation of $g$ completes returning $o$.
+If the evaluation of $e$ throws then
+the invocation of $g$ completes throwing the same object and stack trace,
+and no binding of $v$ to any object occurs.
```

Comment:
Change to:
```
and does not change the binding of $v$.
```
It's possible for the evaluation of the initializer expression to bind `v` and then throw. It's too late to say that no binding occurs. No *further* binding, possibly, but i like the phrase above better.